File: MaximumDepthOfBinaryTree/MaximumDepthOfBinaryTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def maxDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root == None:
            return 0
        else:
            tmp_l = Solution()
            tmp_r = Solution()
            depth_l = tmp_l.maxDepth(root.left)
            depth_r = tmp_r.maxDepth(root.right)
            if depth_l > depth_r:
                return depth_l + 1
            else:
                return depth_r + 1

File: MaximumDepthOfBinaryTree/MaximumDepthOfBinaryTree.cpp
/**
* Definition for a binary tree node.
*/
struct TreeNode {
     int val;
     TreeNode *left;
     TreeNode *right;
     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 };
class Solution {
public:
	int maxDepth(TreeNode* root) {
		int max = 0;
		if (root == nullptr) {
			return max;
		}
		DFS(root, 0, max);
		return max;
	}
private:
	void DFS(TreeNode* root, int count, int& max) {
		if (root != nullptr) {
			DFS(root->left, count + 1, max);
			DFS(root->right, count + 1, max);
		}
		if (count > max) {
			max = count;
		}
	}
};